libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -529,9 +528,32 @@ void AP_ESC_Telem::update()
                 _last_rpm_log_us[i] = _rpm_data[i].last_update_us;
             }
         }
+
+        if ((now_us - _rpm_data[i].last_update_us) > ESC_RPM_DATA_TIMEOUT_US) {
```

Comment:
This is what's actually setting the flag for us, the helper checks the data valid flag, which seemed kinda confusing.

Reply:
Okay, I came back to this, it does work just fine, but it makes it 24 bytes larger. In practice you will still bail out at the same time in general, but it has to run 2 more checks then it otherwise would have if that's not the case. I'm inclined to leave it as I presented since this is the explicit wrap protection.